lib/tools.py

```diff
@@ -160,3 +152,235 @@ def reset_retriever(scope):
+                    previous_list = new_list
+
+        for lines in previous_list:
+            outfile.write("\n" + str(lines[0:-1]))
+    outfile.close()
+    fp.close()
+    os.system(" rm -r {}".format(input_file))
+    return output_file
+
+
+def xml2csv(input_file, outputfile=None, header_values=None, row_tag="row"):
+    """Convert xml to csv
+
+    <root>
+    <row>
+       <User>Alex</User>
+       <Country>US</Country>
+       <Country>PT</Country>
+       <Age>25</Age>
+    </row>
+    <row>
+       <User>Ben</User>
+       <Country>US</Country>S
+       <Age>24</Age>
+    </row>
+    </root>
+
+    User,Country,Age
+    Alex,US,25
+    Alex,Uk,25
+    Ben,US,24
+    """
+    fp = open(input_file, 'r')
+    tree = ET.parse(fp)
+    root = tree.getroot()
+
+    # set output file name and write header
+    if outputfile is None:
+        outputfile = str(os.path.splitext(os.path.basename(input_file))[0]) + ".csv"
+    outfile = open(outputfile, 'w')
+    outfile.write(",".join(header_values))
+
+    # lines in xml
+    for rows in root.findall(row_tag):
+        previous_lists = [""]
+        if header_values:
+            # for each line, extract values for corresponding to column name
+            for column_name in header_values:
+                new_list = []
+                # check if multiple values exist
+                if len(rows.findall(column_name)) > 1:
+                    for child_item in rows.findall(column_name):
+                        # create new list with previous values and new cross-tab values added
+                        for old_lines in previous_lists:
+                            value_x = ""
+
+                            if child_item.text is None:
+                                pass
+                            else:
+                                value_x = str(child_item.text)
+                            temp = str(str(old_lines) + value_x + ",")
+                            new_list.append(temp)
+                    previous_lists = new_list
+                else:
+                    # no multiple values, just add available child
+                    for p_strings in previous_lists:
+
+                        value_x = ""
+                        if rows.find(column_name).text is None:
+                            pass
+                        else:
+                            value_x = str(rows.find(column_name).text)
+                        new_list.append("".join(str(p_strings) + value_x + ","))
+                        previous_lists = new_list
+        else:
+            print ("no header provided")
+            exit()
+        for lines in previous_lists:
+            outfile.write("\n" + str(lines[0:-1]))
+    outfile.close()
+    fp.close()
+    os.system(" rm -r {}".format(input_file))
+    return outputfile
+
+
+def getmd5(path=None, mode='rb', lines=None):
+    """Get MD5 of a file, files in a directory or MD5 value of a set of lines
+
+    default mode 'rb'
```

Comment:
This is already clear from the function definition so I don't think including it in the docstring adds anything. Listing the possible modes would though. e.g. "Possible values for mode include the default, 'rb', and 'rU', which ..."